### DISCUSSION TITLE
Connector Availability and Status relationship

### DISCUSSION THREAD

**Discussion by SiarheiVolkau** [c-jimenez/open-ocpp#140] [Q&A] [answered]
I'm writing this to make things clear, it might/might not reflect real issue in the codebase.

As I know `availability` is designed to be persistent connector attribute, controlled remotely via `ChangeAvailability`.
While connector `status` shall reflect actual status of the physical connector - it is a function of real status and `availability` and it don't have to be persistent.

So the question is: why open-ocpp stores `status` in the database and don't store `availability` anyhow.
I guess, its because `availability` takes precedence and `Status::Unavailable` strictly bidirectionally equal to `Availability::Inoperative`, only in that case it makes sense. Another drawback of that approach is that real connector status can't be `Unavailable` because it overrides `availability` too.

Are my assumptions correct?

**Comment by c-jimenez** [comment, marked as answer]
Hi,

The availability status is not a well-defined status in OCPP 1.6.
The standard says :

```
Central System can request a Charge Point to change its availability. A Charge Point is considered available
(“operative”) when it is charging or ready for charging. A Charge Point is considered unavailable when it does not
allow any charging.
```

So in my implementation I decided to merge what you call the `availability` with the `Unavailable` status of a connector when the Central System ask the connector to be in `Inoperative` state. All the other connector statuses are then reflecting the `Operative` state.

I didn't think it would be a drawback since normally in `Inoperative` state the connector cannot allow any charge and thus cannot be on any other state than `Available` or `Unavailable`.

Is this a blocking point for your Charge Point implementation?
If so, which are your use cases which needs to separate this 2 information?

**Comment by SiarheiVolkau** [comment]
> Is this a blocking point for your Charge Point implementation?
> If so, which are your use cases which needs to separate this 2 information?

Yes, it is. Let me describe the situation.
I have a ChargePoint-like device (CP) and up to 4 Connector-like devices (CNs) connected via LAN.
CP can control power supply of CNs, so there's a period for CN to boot up (usually 2 minutes).
For that period of time it will be good to use `Unavailable` status, since it fit best.
Also, IMHO, it shall remain `Unavailable` if it didn't appear in 2 minutes (physically absent).
That's where the things go wrong.

**Reply by c-jimenez** [reply]
You can control the status of your connectors and force it to stay 'Unavailable' until they are ready by using the `IChargePoint::statusNotification()`.

Is this not sufficient?

**Reply by SiarheiVolkau** [reply]
I do that, although I also want to know `Availability` which is set by CentralSystem for whatever reason.
e.g. CS admin might set connector to `Inoperative` for planning maintenance.
So in my opinion `Status` and `Availability` shall be independent things.

Thank you for the quick answering, now I realized what I have to do.
